### ISSUE TITLE
oc cluster up --metrics still fails in openshift 3.5.5

### ISSUE THREAD

**Issue by fmarchioni** [openshift/origin#14277]
I'm checking if I'm able to start Openshift Origin with metrics enabled. As per https://bugzilla.redhat.com/show_bug.cgi?id=1416240 you should be able to run it with OCP >= 3.5.0 
Here is how to reproduce the failure:
##### Version
oc v3.4.1.2
kubernetes v1.4.0+776c994
features: Basic-Auth GSSAPI Kerberos SPNEGO

Server https://192.168.1.66:8443
openshift v3.5.5.15
kubernetes v1.5.2+43a9be4

##### Steps To Reproduce
```
$ sudo oc cluster up --version=latest --metrics
-- Checking OpenShift client ... OK
-- Checking Docker client ... OK
-- Checking Docker version ... OK
-- Checking for existing OpenShift container ... OK
-- Checking for registry.access.redhat.com/openshift3/ose:latest image ... 
   Pulling image registry.access.redhat.com/openshift3/ose:latest
   Pulled 2/4 layers, 50% complete
   Pulled 3/4 layers, 75% complete
   Pulled 4/4 layers, 100% complete
   Extracting
   Image pull complete
-- Checking Docker daemon configuration ... OK
-- Checking for available ports ... 
   WARNING: Binding DNS on port 8053 instead of 53, which may not be resolvable from all clients.
-- Checking type of volume mount ... 
   Using nsenter mounter for OpenShift volumes
-- Creating host directories ... OK
-- Finding server IP ... 
   Using 192.168.1.66 as the server IP
-- Starting OpenShift container ... 
   Creating initial OpenShift configuration
   Starting OpenShift using container 'origin'
   Waiting for API server to start listening
   OpenShift server started
-- Adding default OAuthClient redirect URIs ... OK
-- Installing registry ... OK
-- Installing router ... OK
-- Installing metrics ... OK
-- Importing image streams ... OK
-- Importing templates ... OK
-- Login to server ... OK
-- Creating initial project "myproject" ... OK
-- Removing temporary directory ... OK
-- Server Information ... 
   OpenShift server started.
   The server is accessible via web console at:
       https://192.168.1.66:8443

   The metrics service is available at:
       https://metrics-openshift-infra.192.168.1.66.xip.io

   You are logged in as:
       User:     developer
       Password: developer

   To login as administrator:
       oc login -u system:admin
```

```
$ oc login -u system:admin
Logged into "https://192.168.1.66:8443" as "system:admin" using existing credentials.
```

```
$ oc get pods -n openshift-infra
NAME                   READY     STATUS    RESTARTS   AGE
metrics-deployer-pod   0/1       Error     0          2m
```

```
$ oc logs metrics-deployer-pod -n openshift-infra
++ parse_bool false CONTINUE_ON_ERROR
++ local v=false
++ '[' false '!=' true -a false '!=' false ']'
++ echo false
+ continue_on_error=false
+ '[' false == false ']'
+ set -eu
+ deployer_mode=deploy
+ image_prefix=registry.access.redhat.com/openshift3/ose-
+ image_version=latest
+ startup_timeout=500
+ master_url=https://kubernetes.default.svc:443
+ [[ 3 == \/ ]]
++ parse_bool false REDEPLOY
++ local v=false
++ '[' false '!=' true -a false '!=' false ']'
++ echo false
+ redeploy=false
+ '[' false == true ']'
+ mode=deploy
+ '[' deploy = redeploy ']'
++ parse_bool false IGNORE_PREFLIGHT
++ local v=false
++ '[' false '!=' true -a false '!=' false ']'
++ echo false
+ ignore_preflight=false
+ cassandra_nodes=1
++ parse_bool false USE_PERSISTENT_STORAGE
++ local v=false
++ '[' false '!=' true -a false '!=' false ']'
++ echo false
+ use_persistent_storage=false
++ parse_bool false DYNAMICALLY_PROVISION_STORAGE
++ local v=false
++ '[' false '!=' true -a false '!=' false ']'
++ echo false
+ dynamically_provision_storage=false
+ cassandra_pv_size=10Gi
+ metric_duration=7
+ user_write_access=false
+ heapster_node_id=nodename
+ metric_resolution=10s
+ project=openshift-infra
+ master_ca=/var/run/secrets/kubernetes.io/serviceaccount/ca.crt
+ token_file=/var/run/secrets/kubernetes.io/serviceaccount/token
+ dir=/etc/deploy/_output
+ secret_dir=/secret
+ rm -rf /etc/deploy/_output
+ mkdir -p /etc/deploy/_output
+ chmod 700 /etc/deploy/_output
+ mkdir -p /secret
+ chmod 700 /secret
chmod: changing permissions of '/secret': Read-only file system
+ :
+ hawkular_metrics_hostname=metrics-openshift-infra.192.168.1.66.xip.io
+ hawkular_metrics_alias=hawkular-metrics
+ hawkular_cassandra_alias=hawkular-cassandra
++ date +%s
+ openshift admin ca create-signer-cert --key=/etc/deploy/_output/ca.key --cert=/etc/deploy/_output/ca.crt --serial=/etc/deploy/_output/ca.serial.txt --name=metrics-signer@1495464871
+ '[' -n 1 ']'
+ oc config set-cluster master --api-version=v1 --certificate-authority=/var/run/secrets/kubernetes.io/serviceaccount/ca.crt --server=https://kubernetes.default.svc:443
cluster "master" set.
++ cat /var/run/secrets/kubernetes.io/serviceaccount/token
+ oc config set-credentials account --token=eyJhbGciOiJSUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJrdWJlcm5ldGVzL3NlcnZpY2VhY2NvdW50Iiwia3ViZXJuZXRlcy5pby9zZXJ2aWNlYWNjb3VudC9uYW1lc3BhY2UiOiJvcGVuc2hpZnQtaW5mcmEiLCJrdWJlcm5ldGVzLmlvL3NlcnZpY2VhY2NvdW50L3NlY3JldC5uYW1lIjoibWV0cmljcy1kZXBsb3llci10b2tlbi0xbnhzMiIsImt1YmVybmV0ZXMuaW8vc2VydmljZWFjY291bnQvc2VydmljZS1hY2NvdW50Lm5hbWUiOiJtZXRyaWNzLWRlcGxveWVyIiwia3ViZXJuZXRlcy5pby9zZXJ2aWNlYWNjb3VudC9zZXJ2aWNlLWFjY291bnQudWlkIjoiN2RlYzJjYzEtM2VmZS0xMWU3LThlNGYtNjhmNzI4ZjNiOTdkIiwic3ViIjoic3lzdGVtOnNlcnZpY2VhY2NvdW50Om9wZW5zaGlmdC1pbmZyYTptZXRyaWNzLWRlcGxveWVyIn0.Ty-dGlac3pn2ohpi083w2tX0vd9elZ1rJsp32S6MZyBIODu6BqdajhmVQj3U0UIsll8gLOoRQqj36OswS29Vd76UEcN5800cJQLG6-CKtvC4oNVuapti1L3p3dA0L37Rw4afXkYZqjCUr28b-zKhy4Yp7rkCP70_59_ZR-TliMzTMqgKUkDgJn14RZznIqNcAb3XwUA38DFAtePazRQIAhRSk2XcxYT5CWG7JejpT4JoM4Al_ZsruxFduqbWRXktqUqKkSooGHnzXoH-osJKYAme_FatPUfd1Ys48SEwQ65Z9QQRamNvcqmUr7WZHvEZFOl6lAKjltszF7L_IGh2Sg
user "account" set.
+ oc config set-context current --cluster=master --user=account --namespace=openshift-infra
context "current" set.
+ oc config use-context current
switched to context "current".
+ old_kc=/etc/deploy/.kubeconfig
+ KUBECONFIG=/etc/deploy/_output/kube.conf
+ '[' -z 1 ']'
+ oc config set-cluster deployer-master --api-version=v1 --certificate-authority=/var/run/secrets/kubernetes.io/serviceaccount/ca.crt --server=https://kubernetes.default.svc:443
cluster "deployer-master" set.
++ cat /var/run/secrets/kubernetes.io/serviceaccount/token
+ oc config set-credentials deployer-account --token=eyJhbGciOiJSUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJrdWJlcm5ldGVzL3NlcnZpY2VhY2NvdW50Iiwia3ViZXJuZXRlcy5pby9zZXJ2aWNlYWNjb3VudC9uYW1lc3BhY2UiOiJvcGVuc2hpZnQtaW5mcmEiLCJrdWJlcm5ldGVzLmlvL3NlcnZpY2VhY2NvdW50L3NlY3JldC5uYW1lIjoibWV0cmljcy1kZXBsb3llci10b2tlbi0xbnhzMiIsImt1YmVybmV0ZXMuaW8vc2VydmljZWFjY291bnQvc2VydmljZS1hY2NvdW50Lm5hbWUiOiJtZXRyaWNzLWRlcGxveWVyIiwia3ViZXJuZXRlcy5pby9zZXJ2aWNlYWNjb3VudC9zZXJ2aWNlLWFjY291bnQudWlkIjoiN2RlYzJjYzEtM2VmZS0xMWU3LThlNGYtNjhmNzI4ZjNiOTdkIiwic3ViIjoic3lzdGVtOnNlcnZpY2VhY2NvdW50Om9wZW5zaGlmdC1pbmZyYTptZXRyaWNzLWRlcGxveWVyIn0.Ty-dGlac3pn2ohpi083w2tX0vd9elZ1rJsp32S6MZyBIODu6BqdajhmVQj3U0UIsll8gLOoRQqj36OswS29Vd76UEcN5800cJQLG6-CKtvC4oNVuapti1L3p3dA0L37Rw4afXkYZqjCUr28b-zKhy4Yp7rkCP70_59_ZR-TliMzTMqgKUkDgJn14RZznIqNcAb3XwUA38DFAtePazRQIAhRSk2XcxYT5CWG7JejpT4JoM4Al_ZsruxFduqbWRXktqUqKkSooGHnzXoH-osJKYAme_FatPUfd1Ys48SEwQ65Z9QQRamNvcqmUr7WZHvEZFOl6lAKjltszF7L_IGh2Sg
user "deployer-account" set.
+ oc config set-context deployer-context --cluster=deployer-master --user=deployer-account --namespace=openshift-infra
context "deployer-context" set.
+ '[' -n 1 ']'
+ oc config use-context deployer-context
switched to context "deployer-context".
+ case $deployer_mode in
+ '[' false '!=' true ']'
+ validate_preflight
+ set +x

PREFLIGHT CHECK FAILED
========================
validate_master_accessible: 
unable to access master url https://kubernetes.default.svc:443
See the error from 'curl https://kubernetes.default.svc:443' below for details:
curl: (6) Could not resolve host: kubernetes.default.svc; Unknown error

Deployment has been aborted prior to starting, as these failures often indicate fatal problems.
Please evaluate any error messages above and determine how they can be addressed.
To ignore this validation failure and continue, specify IGNORE_PREFLIGHT=true.

PREFLIGHT CHECK FAILED
```

##### Current Result
```
$ oc get pods -n openshift-infra
NAME                   READY     STATUS    RESTARTS   AGE
metrics-deployer-pod   0/1       Error     0          2m
```
##### Expected Result
Expecting to find metrics pods running.
##### Additional Information
```
$ oadm diagnostic
Info:  Successfully read a client config file at '/home/francesco/.kube/config'
Info:  Using context for cluster-admin access: 'myproject/192-168-1-66:8443/system:admin'

[Note] Running diagnostic: ConfigContexts[myproject/192-168-42-149:8443/developer]
       Description: Validate client config context is complete and has connectivity
       
ERROR: [DCli0015 from diagnostic ConfigContexts@openshift/origin/pkg/diagnostics/client/config_contexts.go:285]
       For client config context 'myproject/192-168-42-149:8443/developer':
       The server URL is 'https://192.168.42.149:8443'
       The user authentication is 'developer/192-168-42-149:8443'
       The current project is 'myproject'
       (*url.Error) Get https://192.168.42.149:8443/api: dial tcp 192.168.42.149:8443: getsockopt: no route to host
       Diagnostics does not have an explanation for what this means. Please report this error so one can be added.
       
[Note] Running diagnostic: ConfigContexts[myproject/192-168-1-66:8443/system:admin]
       Description: Validate client config context is complete and has connectivity
       
Info:  The current client config context is 'myproject/192-168-1-66:8443/system:admin':
       The server URL is 'https://192.168.1.66:8443'
       The user authentication is 'system:admin/192-168-1-66:8443'
       The current project is 'myproject'
       Successfully requested project list; has access to project(s):
         [default kube-system myproject openshift openshift-infra]

```

**Comment by pweil-** [comment]
@jcantrill this is similar to https://github.com/openshift/origin/issues/11965 but not quite the same as this shows the service not being accessible.

**Comment by fmarchioni** [comment]
btw I have tested also with 3.4 release and passing the image registry at startup with the same result:
```
$ sudo oc cluster up --image=registry.access.redhat.com/openshift3/ose --version=v3.4 --metrics=true
. . . .
$ oc get pods -n openshift-infra
NAME                   READY     STATUS    RESTARTS   AGE
metrics-deployer-pod   0/1       Error     0          43s
```

**Comment by jcantrill** [comment]
Please try v3.6 or better.  Closing as we are unlikely to fix in 1.5/3.5 release of the code.
